src/components/Publish/_utils.ts

```diff
@@ -506,16 +507,27 @@ export async function verifyRawServiceCredential(
       }
     }
     if (response?.status === 201) {
-      const serviceOffering = parsedServiceCredential.verifiableCredential.find(
-        (credential) =>
-          credential?.credentialSubject?.type === 'gx:ServiceOffering'
-      )
-      const credentialId = serviceOffering?.credentialSubject?.id
+      const dependsOn = parsedServiceCredential.verifiableCredential
+        .filter(
+          (credential) =>
+            credential?.credentialSubject['gx:dependsOn'] !== undefined
+        )
+        .map(
+          (credential) => credential?.credentialSubject['gx:dependsOn'][0].id
+        )
+      const serviceOffering =
+        parsedServiceCredential.verifiableCredential.filter(
+          (credential) =>
+            credential?.credentialSubject?.type === 'gx:ServiceOffering' &&
+            credential?.credentialSubject?.id !== dependsOn?.[0]
+        )
+      const credentialId = serviceOffering[0]?.credentialSubject?.id
```

Comment:
issue(blocking): with this approach we always use the first (`[0]`) array entry of `dependsOn` and `serviceOffering`. We need to make sure to evaluate all `serviceOfferings` found in the credentials array.

pseudo code would be something like:
```
serviceOfferings = parsedCredential.filter(type = 'serviceOffering')
dependOnIds = serviceOfferings.map(service.dependsOn)
rootService = serviceOfferings.filter(!dependOnIds.includes(service.id))

if(rootService.length !== 1)
  // could not be verified
else
  // match asset id against rootService id
```

Reply:
suggestion: for easier mapping of arrays in this case (e.g. `gx:dependsOn`), have a look at the [`flatMap()` array method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)